mule-extensions-api/src/main/java/org/mule/runtime/extension/api/util/NameUtils.java

```diff
@@ -257,12 +257,14 @@ public static boolean isUncountable(String word) {
    * @return the hypenized name for the given {@code type}
    */
   public static String getTopLevelTypeName(MetadataType metadataType) {
-    String aliasName = metadataType.getAnnotation(TypeAliasAnnotation.class).map(TypeAliasAnnotation::getValue)
+    return hyphenize(getAliasName(metadataType));
+  }
+
+  public static String getAliasName(MetadataType metadataType) {
+    return metadataType.getAnnotation(TypeAliasAnnotation.class).map(TypeAliasAnnotation::getValue)
         .orElseGet(() -> getTypeId(metadataType).map(typeId -> metadataType.getMetadataFormat().equals(JAVA)
             ? getAliasName(getType(metadataType))
             : typeId).orElseThrow(() -> new IllegalArgumentException("No name available for the given type")));
-
-    return hyphenize(aliasName);
```

Comment:
since Alias is not necessarily hyphenized, we now have a way to obtain in "as it was" instead of automatically hyphenizing it. That logic was moved to the "getTopLevelElementName", which is where we actualy needed the hyphenization